### PR TITLE
Add get method for causes of interruption.

### PR DESCRIPTION
**Pull request by lvotypko**

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
:+1:  (Javadoc can be patched during the merge)

File: core/src/main/java/hudson/model/Executor.java [hunk 1]
```diff
@@ -472,6 +474,16 @@ public void killHard() {
...
         }
     }
+    
+    /**
+     * Returns causes of interruption
+     *
+     * @return Unmodifiable collection of causes of interruption
+     *      
+     */
+    public Collection<CauseOfInterruption> getCausesOfInterruption(){
```
Comment (KostyaSha) [inline core/src/main/java/hudson/model/Executor.java]: `@Nonnull` ?

Comment (olivergondza) [inline core/src/main/java/hudson/model/Executor.java]: `@since` and `.` on the end of sentences in javadoc.

Comment (KostyaSha) [inline core/src/main/java/hudson/model/Executor.java]: @olivergondza and space after `()` then ;)

File: core/src/main/java/hudson/model/Executor.java [hunk 2]
```diff
@@ -472,6 +474,16 @@ public void killHard() {
             lock.readLock().unlock();
         }
     }
+    
+    /**
+     * Returns causes of interruption.
+     *
+     * @return Unmodifiable collection of causes of interruption.
+     * @since  1.617
```
Comment (KostyaSha) [inline core/src/main/java/hudson/model/Executor.java]: not actual now, please replace with TODO

**Comment by KostyaSha** [comment]
Please fix since and squash and then :+1:

**Comment by jglick** [comment]
Seems fine, though is there a specific use case you had in mind?

**Comment by oleg-nenashev** [comment]
Manually integrated as https://github.com/jenkinsci/jenkins/commit/efa1487e85241f7ca604b305bd1f436404cd75c2

**Pull request closed without merging**